gravitational/teleport
# Fix not being able to "login" with auth type set to sso but no connectors set yet

fixes https://github.com/gravitational/teleport/issues/27466
regression starts from `v12.1.3` 

The web UI was calling the `ping` endpoint where out of all the fields given in response, we only utilized flags for `assist` and `automatic upgrades`, but this endpoint did a bunch of other things such as looking up sso connectors. The `/ping` endpoint will return an error if cluster auth setting is set to either github,saml,oidc and no connectors were found.

I wasn't sure if this was already thought of (putting flags into webconfig instead) and there was a reason why we decided to call the public endpoint `ping` after calling private `usercontext`. If there is a reason, there is another way to fix it that keeps the original behavior, but if not, this PR saves a trip to the backend and save doing extra work. 

The web config is retrieved before the user is logged in (if this bit matters)

Also, if this fix is accepted, i am wondering if the flag `assist` and `automatic upgrade` is necessary in the web `ping` response? (tried looking into it and looked like it was only useful for the web UI) We have two ping endpoints where if i understand correctly, should give the same result (just handling of looking up sso connectors is different), and the response has diverged b/c of these flags (the two ping endpoints is used [here](https://github.com/gravitational/teleport/blob/5d53c91c7a8d41391422171749b3a3d7f23084ab/api/client/webclient/webclient.go#L205))

#### Manual testing
I tested that `automatic upgrades` can be enabled starting teleport with flag `TELEPORT_AUTOMATIC_UPGRADES=1` and disabled without the flag

~~I am still working on trying to test out the assist flag~~ tested by jakule 🙏 

To test yourself, start teleport with the flags you want to see enabled/disabled, load the web page, and under networks tab look for `config.js` request.
